pkg/invoke/raw_exec.go

```diff
@@ -16,21 +16,66 @@ package invoke
 
 import (
 	"bytes"
+	"context"
 	"encoding/json"
 	"fmt"
 	"io"
 	"os/exec"
+	"time"
 
 	"github.com/containernetworking/cni/pkg/types"
 )
 
+var ErrorPluginExecTimeout = fmt.Errorf("Waiting for plugin to return timed out")
+
 type RawExec struct {
 	Stderr io.Writer
 }
 
+func (e *RawExec) ExecPluginWithTimeout(pluginPath string, stdinData []byte, environ []string, timeout time.Duration) ([]byte, error) {
+	ctx, cancel := context.WithTimeout(context.TODO(), timeout)
+	defer cancel()
+	stdout := &bytes.Buffer{}
+	cmd, err := e.start(pluginPath, stdinData, stdout, environ)
+	if err != nil {
+		return nil, pluginErr(err, stdout.Bytes())
+	}
+	cmdDone := make(chan error)
+	go func() {
+		cmdDone <- cmd.Wait()
+	}()
+
+	select {
+	case <-ctx.Done():
+		// kill the plugin process
+		if cmd.Process != nil {
+			cmd.Process.Kill()
```

Comment:
you need to still receive the results from cmdDone here, or else you can get a null pointer exception.